### PR TITLE
Add get/set support for the MAC address of taps.

### PR DESCRIPTION
**Pull request by jeamland**

### CODE REVIEW

File: src/address.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
+use std::convert::TryFrom;
+
+#[derive(Clone, Copy)]
+pub struct EthernetAddr([u8; 6]);
```
Comment (yaa110) [inline src/address.rs:4]: Please use [`mac_address`](https://crates.io/crates/mac_address) crate which is stable and cross-platform.

File: src/linux/address.rs [hunk 2]
```diff
@@ -36,3 +38,43 @@ impl Ipv4AddrExt for Ipv4Addr {
         ntoh(sock.sin_addr.s_addr).into()
     }
 }
+
+impl Into<sockaddr> for EthernetAddr {
+    fn into(self) -> sockaddr {
+        let mut addr: [std::os::raw::c_char; 14usize] = [0; 14];
+        addr[..6].copy_from_slice(unsafe { &*(self.as_bytes() as *const [u8] as *const [i8]) });
```
Comment (yaa110) [inline src/linux/address.rs:45]: cast types could be inferred using [`InferredType`](https://doc.rust-lang.org/reference/types/inferred.html):

```rust
unsafe { &*(self.as_bytes() as *const _ as *const _) }
```

File: src/linux/address.rs [hunk 3]
```diff
@@ -36,3 +38,43 @@ impl Ipv4AddrExt for Ipv4Addr {
...
+
+impl TryFrom<sockaddr> for EthernetAddr {
+    type Error = EthernetAddrError;
+
+    fn try_from(addr: sockaddr) -> Result<Self, Self::Error> {
+        if addr.sa_family != libc::ARPHRD_ETHER {
+            return Err(EthernetAddrError::WrongType(addr.sa_family));
+        }
+
+        let mut a: [u8; 6] = [0; 6];
```
Comment (yaa110) [inline src/linux/address.rs:76]: please use a more informative name for letting this variable.

File: src/linux/address.rs [hunk 4]
```diff
@@ -36,3 +38,43 @@ impl Ipv4AddrExt for Ipv4Addr {
...
+        a.copy_from_slice(unsafe { &*(&addr.sa_data[..6] as *const [i8] as *const [u8]) });
```
Comment (yaa110) [inline src/linux/address.rs:77]: please use `InferredType` for casting.

**Review by yaa110 — CHANGES_REQUESTED** [review]
Thank you for your contribution.

File: src/address.rs (hunk 1 above)
Comment (jeamland) [reply src/address.rs]: Done. Thanks for pointing that one out!

File: src/linux/address.rs (hunk 2 above)
Comment (jeamland) [reply src/linux/address.rs]: Done.

File: src/linux/address.rs (hunk 3 above)
Comment (jeamland) [reply src/linux/address.rs]: Done.

File: src/linux/address.rs (hunk 4 above)
Comment (jeamland) [reply src/linux/address.rs]: Done.

**Comment by jeamland** [comment]
Thanks for the feedback! Let me know if there's anything else.

**Pull request merged**